### PR TITLE
Change arrayComponent to extension method.

### PR DESCRIPTION
**Pull request by JakeWharton**

### CODE REVIEW

File: src/main/java/com/squareup/kotlinpoet/TypeName.kt [hunk 1]
```diff
@@ -227,10 +227,10 @@ abstract class TypeName internal constructor(
       }
     }
 
-    /** Returns the array component of `type`, or null if `type` is not an array.  */
-    @JvmStatic fun arrayComponent(type: TypeName): TypeName? {
-      return if (type is ParameterizedTypeName && type.rawType == ARRAY)
-        type.typeArguments.single() else
+    /** Returns the array component of this [TypeName], or null if it is not an array.  */
+    @JvmStatic fun TypeName.arrayComponent(): TypeName? {
```
Comment (swankjesse) [inline src/main/java/com/squareup/kotlinpoet/TypeName.kt]: Weird to use an extension method for a class we declare?

Comment (JakeWharton) [inline src/main/java/com/squareup/kotlinpoet/TypeName.kt]: Oh, true. We do this in JavaPoet as well. Can change.

Comment (JakeWharton) [inline src/main/java/com/squareup/kotlinpoet/TypeName.kt]: Actually I think this used to be on `ArrayTypeName`?

Comment (swankjesse) [inline src/main/java/com/squareup/kotlinpoet/TypeName.kt]: Yep. For Kotlin arrays aren’t special, right?

Comment (JakeWharton) [inline src/main/java/com/squareup/kotlinpoet/TypeName.kt]: Damnit. Yes. I need to start thinking about what I'm doing rather than blindly coding.

**Pull request merged**